bwa: add page (#12720)
tldr-pages/tldr: pages/linux/bwa.md

# bwa

> > BWA is a short, low-divergent DNA sequences mapper against a large reference genome, such as the human genome.

Reviewer: ```suggestion
> Burrows-Wheeler Alignment tool.
> Short, low-divergent DNA sequences mapper against a large reference genome, such as the human genome.
```
How about this? (Referred to their man page at https://bio-bwa.sourceforge.net/bwa.shtml)

Author: Their man page is not completely up to date, but I like this desc.

> More information: <https://github.com/lh3/bwa>.

> - Index reference genome:

Reviewer: ```suggestion
- Index the reference genome:
```

`bwa index {{path/to/reference.fa}}`

> - Map single-end reads (sequences) to indexed genome using 32 [t]*hreads and compress the result to save space:

Reviewer 2: ```suggestion
- Map single-end reads (sequences) to indexed genome using 32 [t]hreads and compress the result to save space:
```

> `bwa mem -t 32 {{path/to/reference.fa}} {{path/to/read_single_end.fq.gz}} | gzip -3 > {{path/to/alignment_single_end.sam.gz}}`

Reviewer 2: We tend not to use commands which involve other commands, however given the complexity of the command, it might be a good idea to see what others think.

Reviewer 3: Is it really necessary to specify the `-3` in the gzip command? If we left it out, the command would be more concise.

Author: I don't mind the change. this `-3` came from their examples posted in bwa repo.

> `bwa mem -t 32 {{path/to/reference.fa}} {{path/to/read_single_end.fq.gz}} | gzip -3 > {{path/to/alignment_single_end.sam.gz}}`
> 
> - Map pair-end reads (sequences) to indexed genome using 32 [t]hreads and compress the result to save space:
> 
> `bwa mem -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip -3 > {{path/to/alignment_pair_end.sam.gz}}`
> 
> - Map pair-end reads (sequences) to indexed genome using 32 [t]hreads with [m]arking shorter split hits as secondary for output SAM file compatibility in Picard software and compress the result:
> 
> `bwa mem -M -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip -3 > {{path/to/alignment_pair_end.sam.gz}}`
> 
> - Map pair-end reads (sequences) to indexed genome using 32 [t]hreads with FASTA/Q [c]omments (e.g. BC:Z:CGTAC) appending to a compressed result:
> 
> `bwa mem -C -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip -3 > {{path/to/alignment_pair_end.sam.gz}}`

Reviewer: ```suggestion
`bwa mem -t 32 {{path/to/reference.fa}} {{path/to/read_single_end.fq.gz}} | gzip > {{path/to/alignment_single_end.sam.gz}}`

- Map pair-end reads (sequences) to the indexed genome using 32 [t]hreads and compress the result to save space:

`bwa mem -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip > {{path/to/alignment_pair_end.sam.gz}}`

- Map pair-end reads (sequences) to the indexed genome using 32 [t]hreads with [m]arking shorter split hits as secondary for output SAM file compatibility in Picard software and compress the result:

`bwa mem -M -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip > {{path/to/alignment_pair_end.sam.gz}}`

- Map pair-end reads (sequences) to indexed genome using 32 [t]hreads with FASTA/Q [c]omments (e.g. BC:Z:CGTAC) appending to a compressed result:

`bwa mem -C -t 32 {{path/to/reference.fa}} {{path/to/read_pair_end_1.fq.gz}} {{path/to/read_pair_end_2.fq.gz}} | gzip > {{path/to/alignment_pair_end.sam.gz}}`
```
